### PR TITLE
Living armor2

### LINKED ISSUE #979: Living Armor Only
**Description**
Obsidimen and may be others are limited to wear living armor only. to enable this, Namegiver items shall get an option "living armor only" and Armor items shall get an option "living armor" 

a small function shall throw a error on the switch item status feature if the namegiver setting is true and the armor is not living.

**Additional context**

### PR DESCRIPTION
**Pull request by patrickmohrmann**
resolves #979  and replaces #994

### CODE REVIEW

File: module/documents/actor.mjs [hunk 1]
```diff
@@ -858,18 +864,23 @@ export default class ActorEd extends Actor {
...
-          const bowAllowed = equippedWeapons[0]?.system.isTwoHandedRanged && itemToUpdate.system.bowUsage;
-          // If there's a two-handed weapon or two one-handed weapons, unequip one
-          const unequipSomeWeapon = equippedWeapons.some( weapon => weapon.system.itemStatus === "twoHands" ) || equippedWeapons.length > 1;
-          if ( !bowAllowed && unequipSomeWeapon ) {
-            // Prefer to unequip off-hand weapon, if available
-            const weaponToUnequip = equippedWeapons.find( weapon => weapon.system.itemStatus === "offHand" ) || equippedWeapons[0];
-            updates.push( { _id: weaponToUnequip.id, "system.itemStatus": "carried" } );
+          // check if namegiver item allows only living armor/shields
+          if ( namegiver.system.livingArmorOnly && itemToUpdate.system.isLivingArmor === false ) {
+            ui.notifications.warn( game.i18n.localize( "ED.Notifications.Warn.livingArmorOnly" ) );
```
Comment (ChristopherSD) [inline module/documents/actor.mjs:870]: See above

File: module/documents/actor.mjs [hunk 2]
```diff
@@ -858,18 +864,23 @@ export default class ActorEd extends Actor {
...
+            break;
+          } else {
```
Comment (ChristopherSD) [inline module/documents/actor.mjs:872]: See above

File: module/data/item/armor.mjs [hunk 3]
```diff
@@ -3,102 +3,108 @@ import ItemDescriptionTemplate from "./templates/item-description.mjs";
...
+      } ),
+      initiativePenalty: new fields.NumberField( {
+        required: true,
+        nullable: false,
+        min:      0,
+        initial:  0,
+        integer:  true,
+        label:    "ED.Item.Armor.initiativePenalty"
+      } ),
+      isLivingArmor: new fields.BooleanField( {
```
Comment (ChristopherSD) [inline module/data/item/armor.mjs:67]: Sorry, yet another suggestion 😅 could we just name it `living`? the item already is armor, so we don't need to repeat item types in properties (this should also apply to things like "piecemealArmor" actually..)
It's always difficult to find a good compromise between descriptive property names and well legible code... But I would suggest that in general: properties don't need information about the object their part of in it...

**Review by ChristopherSD — CHANGES_REQUESTED** [review]

File: module/documents/actor.mjs [hunk 4]
```diff
@@ -797,40 +798,45 @@ export default class ActorEd extends Actor {
...
-              // A complete set of piecemeal armor can have up to 5 size points. Armor pieces come in three sizes and
-              // cost a corresponding number of points: large (3), medium (2), and small (1). A set of piecemeal armor
-              // cannot have more than one size of a particular type.
-              const equippedArmor = this.itemTypes.armor.filter( armor => armor.system.itemStatus === "equipped" );
-              const sameSizePiece = equippedArmor.find( armor => armor.system.piecemealArmor.size === itemToUpdate.system.piecemealArmor.size );
-              if ( sameSizePiece ) {
-                updates.push( { _id: sameSizePiece.id, "system.itemStatus": "carried" } );
+          // check if namegiver item allows only living armor/shields
+          if ( namegiver.system.livingArmorOnly && itemToUpdate.system.isLivingArmor === false ) {
+            ui.notifications.warn( game.i18n.localize( "ED.Notifications.Warn.livingArmorOnly" ) );
```
[the remark this answers is not in the document]
Comment (patrickmohrmann) [reply module/documents/actor.mjs:804]: good idea!

File: module/data/item/armor.mjs (hunk 3 above)
Comment (patrickmohrmann) [reply module/data/item/armor.mjs:67]: perfect!

**Review by ChristopherSD — APPROVED** [review]

**Pull request merged**